File: cpp_package/examples/get_data/src/multiple_streamers.cpp
#include <iostream>
#include <stdlib.h>
#include <string>

#ifdef _WIN32
#include <windows.h>
#else
#include <unistd.h>
#endif

#include "board_shim.h"


int main (int argc, char *argv[])
{
    BoardShim::enable_dev_board_logger ();

    struct BrainFlowInputParams params;
    int board_id = (int)BoardIds::SYNTHETIC_BOARD;
    int res = 0;

    BoardShim *board = new BoardShim (board_id, params);

    try
    {
        board->prepare_session ();
        board->add_streamer ("file://streamer_default_1.csv:w");
        board->add_streamer ("file://streamer_default_2.csv:w");
        board->add_streamer (
            "file://streamer_aux_1.csv:w", (int)BrainFlowPresets::AUXILIARY_PRESET);
        board->add_streamer (
            "file://streamer_aux_2.csv:w", (int)BrainFlowPresets::AUXILIARY_PRESET);
        board->start_stream ();

#ifdef _WIN32
        Sleep (5000);
#else
        sleep (5);
#endif
        board->delete_streamer (
            "file://streamer_aux_2.csv:w", (int)BrainFlowPresets::AUXILIARY_PRESET);

#ifdef _WIN32
        Sleep (5000);
#else
        sleep (5);
#endif

        board->stop_stream ();
        board->release_session ();
    }
    catch (const BrainFlowException &err)
    {
        BoardShim::log_message ((int)LogLevels::LEVEL_ERROR, err.what ());
        res = err.exit_code;
        if (board->is_prepared ())
        {
            board->release_session ();
        }
    }

    delete board;

    return res;
}

File: python_package/examples/tests/multiple_streamers.py
import time

from brainflow.board_shim import BoardShim, BrainFlowInputParams, BoardIds, BrainFlowPresets
from brainflow.data_filter import DataFilter


def main():
    BoardShim.enable_dev_board_logger()

    # use synthetic board for demo
    params = BrainFlowInputParams()
    board_id = BoardIds.SYNTHETIC_BOARD.value

    presets = BoardShim.get_board_presets(board_id)
    print (presets)
    
    board = BoardShim(board_id, params)
    board.prepare_session()
    board.add_streamer('file://streamer_default.csv:w')
    board.add_streamer('streaming_board://225.1.1.1:6677')
    board.start_stream()
    board.add_streamer('streaming_board://225.1.1.1:6678', BrainFlowPresets.AUXILIARY_PRESET)
    board.add_streamer('file://streamer_aux.csv:w', BrainFlowPresets.AUXILIARY_PRESET)
    time.sleep(10)
    board.delete_streamer('streaming_board://225.1.1.1:6678', BrainFlowPresets.AUXILIARY_PRESET)
    board.delete_streamer('streaming_board://225.1.1.1:6677')
    time.sleep(5)
    data_default = board.get_board_data(preset=BrainFlowPresets.DEFAULT_PRESET)
    data_aux = board.get_board_data(preset=BrainFlowPresets.AUXILIARY_PRESET)
    board.stop_stream()
    board.release_session()
    DataFilter.write_file(data_default, 'default.csv', 'w')
    DataFilter.write_file(data_aux, 'aux.csv', 'w')


if __name__ == "__main__":
    main()
